src/zcash/Note.cpp

```diff
@@ -204,14 +205,36 @@ boost::optional<SaplingNotePlaintext> SaplingNotePlaintext::decrypt(
 
     assert(ss.size() == 0);
 
+    uint256 pk_d;
+    if (!librustzcash_ivk_to_pkd(ivk.begin(), ret.d.data(), pk_d.begin())) {
+        return boost::none;
+    }
+
+    uint256 cmu_expected;
+    if (!librustzcash_sapling_compute_cm(
+        ret.d.data(),
+        pk_d.begin(),
+        ret.value(),
+        ret.rcm.begin(),
+        cmu_expected.begin()
+    ))
+    {
+        return boost::none;
+    }
```

Comment:
It was hard to determine whether this is correctly checking that `rcm` is in range, as the spec requires. (The code is distributed between librustzcash and ``Fs::is_valid`` in sapling-crypto.) It is, but more comments might have helped.